RigThread.cpp

```diff
@@ -33,17 +33,28 @@ void RigThread::setup(rig_model_t rig_model, std::string rig_file, int serial_ra
 };
 
 void RigThread::run() {
-    int retcode, status;
+    int retcode;
 
     SoapySDR_log(SOAPY_SDR_DEBUG, "Rig thread starting.");    
 
     rig = rig_init(rigModel);
+    if (rig == NULL) {
+        SoapySDR_log(SOAPY_SDR_ERROR, "Rig failed to init.");
+        terminated.store(true);
+        return;
+    }
+
 	strncpy(rig->state.rigport.pathname, rigFile.c_str(), FILPATHLEN - 1);
-	rig->state.rigport.parm.serial.rate = serialRate;
+    if (serialRate > 0)  // bypass for rate 0 used for USB controlled radios
+	    rig->state.rigport.parm.serial.rate = serialRate;
+
 	retcode = rig_open(rig);
-    
     if (retcode != 0) {
-        SoapySDR_log(SOAPY_SDR_ERROR, "Rig failed to init.");
+        char s[BUFSIZ];
+        snprintf(s, BUFSIZ, "Rig failed to open, rig: %d rig_rate: %d "
+            "rig_port: %s error: %s.", 
+            rigModel, serialRate, rigFile.c_str(), rigerror(retcode));
+        SoapySDR_log(SOAPY_SDR_ERROR, s);
```

Comment:
if you do SoapySDR_logf() you can use the format string and args from the snprintf